libavfilter/dnn/dnn_backend_common.c

```diff
@@ -79,9 +82,11 @@ static void *async_thread_routine(void *args)
     DNNAsyncExecModule *async_module = args;
     void *request = async_module->args;
 
-    async_module->start_inference(request);
+    if (async_module->start_inference(request) != DNN_SUCCESS) {
+        return DNN_ASYNC_FAIL;
+    }
     async_module->callback(request);
-    return NULL;
+    return DNN_ASYNC_SUCCESS;
```

Comment:
the return type is "void *", and where do we check the return value of this function?

Reply:
Yes, on line 26 I have defined this status value as follows. The return value is checked on Line 117 before starting the next inference.

```C
#define DNN_ASYNC_SUCCESS (void *)0
```